File: backend/internal/telegram/mock_client.go
package telegram

import (
	"fmt"
	"log"
)

// MockClient мок-реализация Telegram клиента для разработки и тестирования
type MockClient struct {
	enabled bool
}

// NewMockClient создает новый мок-клиент
func NewMockClient() *MockClient {
	return &MockClient{
		enabled: true,
	}
}

// SendMessage имитирует отправку сообщения в Telegram
// В мок-режиме просто логирует сообщение
func (m *MockClient) SendMessage(token, chatID, message string) error {
	if !m.enabled {
		return fmt.Errorf("mock client is disabled")
	}

	// Логируем вместо реальной отправки
	log.Printf("[MOCK TELEGRAM] Sending message to chat_id=%s: %s", chatID, message)
	log.Printf("[MOCK TELEGRAM] Token: %s...%s", token[:min(10, len(token))], token[max(0, len(token)-5):])

	return nil
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: backend2/src/Telegram/MockClient.php
<?php

namespace App\Telegram;

use Psr\Log\LoggerInterface;

class MockClient implements ClientInterface
{
    public function __construct(
        private readonly LoggerInterface $logger,
        private bool $enabled = true
    ) {}

    public function sendMessage(string $token, string $chatId, string $message): bool
    {
        if (!$this->enabled) {
            $this->logger->error('Mock client is disabled');
            return false;
        }

        // Логируем вместо реальной отправки
        $this->logger->info('[MOCK TELEGRAM] Sending message', [
            'chat_id' => $chatId,
            'message' => $message,
            'token' => substr($token, 0, 10) . '...',
        ]);

        return true;
    }

    public function setEnabled(bool $enabled): void
    {
        $this->enabled = $enabled;
    }
}
